### ISSUE TITLE
Cannot unregister statusChangedCallback ?

### ISSUE THREAD

**Issue by a18250** [Azure/azure-iot-sdk-java#250]
- **OS and version used:** <Android 7.0.1>

- **Java runtime used:** <1.8>

- **SDK version used:** <1.10.0>

# Description of the issue:
The DeviceClient API registerConnectionStatusChangeCallback() and subscribeToDeviceMethod() do not allow null callback as input and there is no corresponding unregister/unsubscribe API.  So how could a client cancel the callback so to stop being called?

# Code sample exhibiting the issue:
N/A

# Console log of the issue:
N/A

**Comment by timtay-microsoft** [comment]
Currently you cannot unregister either of those. We will take a look into allowing that.

**Comment by a18250** [comment]
Okey, thank you much for quick response!

**Comment by timtay-microsoft** [comment]
If you'd like to submit a PR for this feature, I'd be happy to review it for you. You can follow the pattern set by how the old connection status callback allows you to unset it.

**Comment by a18250** [comment]
Hi @timtay-microsoft ,

"old connection status callback" means registerConnectionStateCallback()? How to unregister from it ?

**Comment by timtay-microsoft** [comment]
Yes, registerConnectionStateCallback().

If you edit the code such that the deviceclient does not throw when the callback is null, and the IotHubTransport class to do the same, it will allow you to unset the listener.

**Comment by a18250** [comment]
That will be changes to your requirement and test cases, and to the official doc @ https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.sdk.iot.device._device_client

What is the proper way for a contributor to change all of them?

**Comment by timtay-microsoft** [comment]
Those docs are generated from the javadocs in the source code. All you need to do is add a note in the javadoc for that API that describes how to unset the listener.

For this code change, all that you should need to do is remove the unit tests that check for an IllegalArgumentException when the null callback is registered. You shouldn't have to add any tests.

**Comment by timtay-microsoft** [comment]
If you are looking for a quick fix, you can just register a new connection status change callback that does nothing upon being fired:

```java
  client.registerConnectionStatusChangeCallback(new IotHubConnectionStatusChangeCallback()
  {
      @Override
      public void execute(IotHubConnectionStatus status, IotHubConnectionStatusChangeReason statusChangeReason, Throwable throwable, Object callbackContext) 
      {
          //do nothing
      }
  }, new Object());
```

This will be functionally the same as if you had unset the listener.

**Comment by a18250** [comment]
Register a dummy callback wouldn't help as the reason that we want to unregister is to remove client reference at your SDK side so the client objects can be GCed.

**Comment by timtay-microsoft** [comment]
Sorry for the delay on this feature. The fix for this issue has been checked into master now, and is slated to be released on 8/30.

**Comment by timtay-microsoft** [comment]
The fix for this issue has been released as of Device Client 1.14.0, so I'm closing this issue now.

**Comment by az-iot-builder-01** [comment]
@a18250, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
